### PR TITLE
Add customizable no-retry HTTP status codes (close #297)

### LINKED ISSUE #297: Add customizable no-retry HTTP status codes
Add the ability to define whether to retry or not for a given HTTP status code

### PR DESCRIPTION
**Pull request by Jack-Keene**
This PR adds the ability to add custom retry codes to the emitter, to either retry on failure, or ignore specific codes.

### CODE REVIEW

**Comment by snowplowcla** [comment]
Thanks for your pull request. Is this your first contribution to a Snowplow open source project? Before we can look at your pull request, you'll need to sign a Contributor License Agreement (CLA).

:memo: Please visit https://docs.snowplowanalytics.com/docs/contributing/contributor-license-agreement/ to learn more and sign.

Once you've signed, please reply here (e.g. I signed it!) and we'll verify. Thanks.

File: snowplow_tracker/emitter_configuration.py [hunk 1]
```diff
@@ -144,4 +150,40 @@ def buffer_capacity(self, value: Optional[int]):
...
+
+    @property
+    def custom_retry_codes(self) -> dict:
+        """
+            Custom retry rules for HTTP status codes received in emit responses from the Collector.
+        """
+        return self._custom_retry_codes
+
+    @custom_retry_codes.setter
+    def custom_retry_codes(self, value: Optional[dict]):
```
Comment (matus-tomlein) [inline snowplow_tracker/emitter_configuration.py:163]:
```suggestion
    def custom_retry_codes(self, value: Dict[int, bool]):
```

File: snowplow_tracker/emitter_configuration.py [hunk 2]
```diff
@@ -144,4 +150,40 @@ def buffer_capacity(self, value: Optional[int]):
             raise ValueError("buffer_capacity must greater than 0")
         if not isinstance(value, int) and value is not None:
             raise ValueError("buffer_capacity must be of type int")
-        self._buffer_capacity = value
\ No newline at end of file
+        self._buffer_capacity = value
+
+    @property
+    def custom_retry_codes(self) -> dict:
```
Comment (matus-tomlein) [inline snowplow_tracker/emitter_configuration.py:156]:
```suggestion
    def custom_retry_codes(self) -> Dict[int, bool]:
```

File: snowplow_tracker/emitters.py [hunk 3]
```diff
@@ -71,6 +71,7 @@ def __init__(
         request_timeout: Optional[Union[float, Tuple[float, float]]] = None,
         max_retry_delay_seconds: int = 60,
         buffer_capacity: int = 10000,
+        custom_retry_codes: dict = {}
```
Comment (matus-tomlein) [inline snowplow_tracker/emitters.py:74]:
```suggestion
        custom_retry_codes: Dict[int, bool] = {}
```

**Review by matus-tomlein — APPROVED** [review]
LGTM! 👍

**Review by matus-tomlein — APPROVED** [review]

**Pull request merged**